pkg/resource/user/hooks.go

```diff
@@ -234,6 +234,11 @@ func (rm *resourceManager) syncInlinePolicies(
 		}
 	}
 	for _, pair := range toDelete {
+		// do not remove elements we just updated with `addInlinePolicy`
```

Comment:
`toDelete` consists of those pairs that are existing and not desired; it is impossible that any of these will also be in `toAdd`
`toAdd` consists of those pairs that are desired and not existing; it is impossible that any of these will also be in `toDelete`

If I understand correctly, the corner case this is detecting is when there is a pair in `toDelete` whose key matches another pair that is in `toAdd`; the reason this can be true is that the *value* is different, because this is an update operation. correct?

Reply:
> it is impossible that any of these will also be in toAdd.

It's just `lo.Difference` that is returning the same element in both `toAdd` and `toRemove` (when the element is updated). We need avoid calling delete on a element that just have been added.

Reply:
confirmed offline that what I was describing above is indeed what is going on here